tests/unit_tests/data_steward/resources_test.py

```diff
@@ -65,3 +68,76 @@ def test_cdm_schemas(self):
             result_vocab_tables,
             msg='Vocabulary tables %s should not be in result of cdm_schemas()'
             % result_vocab_tables)
+
+    def test_fields_for(self):
+        """
+        Testing that fields for works as expected with sub-directory structures.
+        """
+        # preconditions
+
+        # test
+        actual_fields = resources.fields_for('person')
+
+        # post conditions
+        person_path = os.path.join(resources.base_path, 'resource_files',
+                                   'fields', 'person.json')
+        with open(person_path, 'r') as fp:
+            expected_fields = json.load(fp)
+
+        self.assertEqual(actual_fields, expected_fields)
+
+    @mock.patch('resources.os.walk')
+    def test_fields_for_duplicate_files(self, mock_walk):
+        """
+        Testing that fields for works as expected with sub-directory structures.
+        """
+        # preconditions
+        walk_results = [
+            (os.path.join('foo', 'bar'), [], ['duplicate.json',
+                                              'unique1.json']),
+            (os.path.join('foo', 'bar',
+                          'baz'), [], ['duplicate.json', 'unique2.json'])
+        ]
+
+        mock_walk.return_value = walk_results
+
+        # test
+        self.assertRaises(RuntimeError, resources.fields_for, 'duplicate')
+
+        # test
+        data = '[{"id": "fake id desc", "type": "fake type"}]'
+        json_data = json.loads(data)
+        with mock.patch('resources.open',
+                        mock.mock_open(read_data=data)) as mock_file:
+            with mock.patch('resources.json.load') as mock_json:
+                mock_json.return_value = json_data
+                fake_path = 'baz'
+                actual_fields = resources.fields_for('duplicate', fake_path)
+                self.assertEqual(actual_fields, json_data)
+
+    @mock.patch('resources.os.walk')
+    def test_fields_for_duplicate_files_with_defined_path(self, mock_walk):
+        """
+        Testing that fields for works as expected with sub-directory structures.
+        """
+        # preconditions
+        walk_results = [[
+            os.path.join('foo', 'bar', 'baz'), '',
+            ['duplicate.json', 'unique2.json']
+        ]]
+
+        mock_walk.return_value = walk_results
+
+        data = '[{"id": "fake id desc", "type": "fake type"}]'
+        json_data = json.loads(data)
+
+        # test
```

Comment:
How is this test different than the last test? 
`fake_path = 'baz'` vs. `fake_path = os.path.join('foo', 'bar', 'baz')`
Is `fake_path = 'baz'` not considered a defined path?

Reply:
I think you're right.  This test isn't providing useful data.

Reply:
This was the original test for testing the sub-directory path, but I copied to contents to the previous unit test because it helped illustrate how we could have multiple files with duplicate names if we specify the sub-directory filepath.  I'll remove this one now.